### PR TITLE
Bibek changes

### PR DESCRIPTION
**Pull request by Aryal007**
updated conf files to input 15 channels and output 3 channels (clean ice, debris, background) for multilabel classification
reducelronplateau changed for 10 steps without improvement in val loss
added label smoothing capability in dice loss
log metrices in a slightly different way to reduce space on tensorboard dashboard
log static images and labels only once 
add script to predict all images in test set
log images only for save_every epochs (reduce tensorboard run file size)

### CODE REVIEW

File: glacier_mapping/models/frame.py [hunk 1]
```diff
@@ -49,7 +48,7 @@ def __init__(self, loss_fn=None, model_opts=None, optimizer_opts=None,
         optimizer_def = getattr(torch.optim, optimizer_opts.name)
         self.optimizer = optimizer_def(self.model.parameters(), **optimizer_opts.args)
         self.lrscheduler = ReduceLROnPlateau(self.optimizer, "min",
-                                             verbose=True, patience=500,
+                                             verbose=True, patience=10,
```
Comment (krisrs1128) [inline glacier_mapping/models/frame.py:51]: Did you mean to use patience=10? This seems like a low number to be using.

File: glacier_mapping/train.py [hunk 2]
```diff
@@ -159,8 +159,9 @@ def log_images(writer, frame, batch, epoch, stage="train"):
     y = torch.flatten(y.permute(0, 1, 3, 2), start_dim=2)
     y_hat = torch.flatten(y_hat.permute(0, 1, 3, 2), start_dim=2)
 
-    writer.add_image(f"{stage}/x", make_grid(pm(squash(x[:, :, :, :3]))), epoch)
-    writer.add_image(f"{stage}/y", make_grid(y.unsqueeze(1)), epoch)
+    if epoch == 1:
```
Comment (krisrs1128) [inline glacier_mapping/train.py]: do you mean epoch == 0?

File: glacier_mapping/models/frame.py (hunk 1 above)
Comment (krisrs1128) [inline glacier_mapping/models/frame.py:51]: I thought I just commented on this -- but isn't this is a small value to use for patience?

File: scripts/train.py [hunk 3]
```diff
@@ -78,18 +78,21 @@
     out_dir = f"{data_dir}/runs/{args.run_name}/models/"
     mask_names = conf.log_opts.mask_names
 
-    for epoch in range(args.epochs):
+    for epoch in range(1,args.epochs+1):
```
Comment (krisrs1128) [inline scripts/train.py]: Why does epoch need to be 1-indexed?

**Review by krisrs1128 — CHANGES_REQUESTED** [review]
A few small comments, but otherwise looks good to me.

File: glacier_mapping/models/frame.py (hunk 1 above)
Comment (Aryal007) [reply glacier_mapping/models/frame.py:51]: Yes, since we only train for 250 epochs max and the model diverges quickly with a high learning rate after a certain point, smaller patience made more sense. However, I haven't explored all the options here.

File: scripts/train.py (hunk 3 above)
Comment (Aryal007) [reply scripts/train.py]: since we were saving on every save_every epoch, using 0-epochs would mean we saved on save_every+1 epoch actually.

File: glacier_mapping/train.py (hunk 2 above)
Comment (Aryal007) [reply glacier_mapping/train.py]: This is to accommodate the previous 1-epoch+1 range. If we decide to use 0-epoch range, this can be 0.

File: glacier_mapping/models/frame.py (hunk 1 above)
Comment (krisrs1128) [reply glacier_mapping/models/frame.py:51]: Ah, okay I was still thinking of your long-running example. No need to explore different parameters here, I just wanted to make sure it wasn't a mistake.

File: scripts/train.py (hunk 3 above)
Comment (krisrs1128) [reply scripts/train.py]: Ah, that's true, that flag is confusing. I think it'd be clearer to just change the save check -- (epoch + 1) % save_every == 0. After that change is made, I'm happy to see this merged in.

Comment (Aryal007) [reply scripts/train.py]: Hi Kris, I made these changes, however, there seems to be a merge conflict with the file.

**Review by krisrs1128 — APPROVED** [review]
